pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -412,30 +434,24 @@ func threadID(didCommMsg *service.DIDCommMsg) (string, error) {
 }
 
 func (s *Service) currentState(thid string) (state, error) {
-	conn, err := s.connectionStore.GetConnectionRecord(thid)
+	connRec, err := s.connectionStore.GetConnectionRecordByNSThreadID(thid)
 	if err != nil {
 		if errors.Is(err, storage.ErrDataNotFound) {
 			return &null{}, nil
 		}
 		return nil, fmt.Errorf("cannot fetch state from store: thid=%s err=%s", thid, err)
 	}
-	return stateFromName(conn.State)
+	return stateFromName(connRec.State)
 }
 
-func (s *Service) update(thid string, state state) error {
-	// todo will be refactored in the issue-397
-	connRecBytes, err := json.Marshal(&ConnectionRecord{State: state.Name(), ThreadID: thid,
-		ConnectionID: generateRandomID()})
-	if err != nil {
-		return err
+func (s *Service) update(msgType string, connectionRecord *ConnectionRecord) error {
+	if msgType == ConnectionRequest && connectionRecord.State == stateNameRequested {
+		return s.connectionStore.saveNewConnectionRecord(connectionRecord)
 	}
-
-	// todo following function will be replaced by persistence connection store save connection record
-	err = s.store.Put("conn_"+thid, connRecBytes)
-	if err != nil {
-		return fmt.Errorf("failed to write to store: %s", err)
+	if msgType == ConnectionInvite && connectionRecord.State == stateNameInvited {
```

Comment:
you can combine the two with OR statement

Reply:
done